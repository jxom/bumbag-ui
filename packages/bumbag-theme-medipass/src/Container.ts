import { css } from 'bumbag';

export default {
  styles: {
    base: props => css`
      ${props.isFluid &&
        css`
          margin: 0px;
        `};
    `
  }
};
