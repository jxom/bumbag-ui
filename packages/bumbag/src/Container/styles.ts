import { css, cssClass } from '../styled';
import { theme } from '../utils';

export const Container = (styleProps) => cssClass`
  width: 100%;

  ${
    !styleProps.isFluid &&
    css`
      max-width: ${theme('breakpoints', 'fullHD')(styleProps)}px;
    `
  };

  ${
    styleProps.isFluid &&
    css`
      padding-left: ${theme(styleProps.themeKey, 'fluidMargin')(styleProps)}rem;
      padding-right: ${theme(styleProps.themeKey, 'fluidMargin')(styleProps)}rem;

      & {
        ${theme(styleProps.themeKey, 'styles.fluid')(styleProps)};
      }
    `
  };

  ${
    (styleProps.isLayout || styleProps.isFluid) &&
    css`
      @media (max-width: ${theme('breakpoints.tablet')(styleProps)}px) {
        padding-left: ${theme(styleProps.themeKey, 'tabletMargin')(styleProps)}rem;
        padding-right: ${theme(styleProps.themeKey, 'tabletMargin')(styleProps)}rem;
      }
    `
  };

  ${
    styleProps.isLayout &&
    css`
      & {
        ${theme(styleProps.themeKey, 'styles.layout')(styleProps)};
      }
    `
  };

  ${getResponsiveProperties(styleProps)};
  ${styleProps.align && !styleProps.isFluid ? alignProperties[styleProps.align] : null};

  & {
    ${theme(styleProps.themeKey, 'styles.base')(styleProps)};
  }
`;

export const alignProperties = {
  left: css`
    margin-right: auto;
  `,
  center: css`
    margin-left: auto;
    margin-right: auto;
  `,
  right: css`
    margin-left: auto;
  `,
};

export function getResponsiveProperties(styleProps) {
  const { breakpoint, isFluid } = styleProps;
  if (isFluid) return;
  if (breakpoint) {
    return css`
      & {
        max-width: ${theme('breakpoints', breakpoint)(styleProps)}px;
      }
    `;
  }
  return css`
    @media (max-width: ${theme('breakpoints', 'fullHD')(styleProps) + 128}px) {
      max-width: ${theme('breakpoints', 'widescreen')(styleProps)}px;
    }

    @media (max-width: ${theme('breakpoints', 'widescreen')(styleProps) + 128}px) {
      max-width: ${theme('breakpoints', 'desktop')(styleProps)}px;
    }

    @media (max-width: ${theme('breakpoints', 'desktop')(styleProps) + 128}px) {
      max-width: ${theme('breakpoints', 'tablet')(styleProps)}px;
    }
  `;
}
