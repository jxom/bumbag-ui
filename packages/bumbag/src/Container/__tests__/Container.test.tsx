import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Container } from '../Container';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Container>Hello world</Container>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Container color="primary">Hello world</Container>);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('alignment', () => {
    it('renders correctly for a left aligned container', () => {
      const { container } = render(<Container align="left">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correctly for a center aligned container', () => {
      const { container } = render(<Container align="center">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correctly for a right aligned container', () => {
      const { container } = render(<Container align="right">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('fixed width', () => {
    it('renders correctly when breakpoint is fullHD', () => {
      const { container } = render(<Container breakpoint="fullHD">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correctly when breakpoint is widescreen', () => {
      const { container } = render(<Container breakpoint="widescreen">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correctly when breakpoint is desktop', () => {
      const { container } = render(<Container breakpoint="desktop">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correctly when breakpoint is tablet', () => {
      const { container } = render(<Container breakpoint="tablet">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correctly when breakpoint is mobile', () => {
      const { container } = render(<Container breakpoint="mobile">test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('fluid', () => {
    it('renders correctly for a fluid container', () => {
      const { container } = render(<Container isFluid>test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('layout', () => {
    it('renders correctly for a layout container', () => {
      const { container } = render(<Container isLayout>test</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Container use="p">Hello world</Container>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Container props', () => {
      const { result } = renderHook(() => Container.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Container>{(ContainerProps) => <div {...ContainerProps}>Hello world</div>}</Container>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Container.base should render correctly', () => {
    const { container } = render(
      <Container overrides={{ Container: { styles: { base: { backgroundColor: 'red' } } } }}>hello world</Container>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.fluid should render correctly', () => {
    const { container } = render(
      <Container isFluid overrides={{ Container: { styles: { fluid: { backgroundColor: 'red' } } } }}>
        hello world
      </Container>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.layout should render correctly', () => {
    const { container } = render(
      <Container isLayout overrides={{ Container: { styles: { layout: { backgroundColor: 'red' } } } }}>
        hello world
      </Container>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.fluidMargin should render correctly', () => {
    const { container } = render(
      <Container isFluid overrides={{ Container: { fluidMargin: 3 } }}>
        hello world
      </Container>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.tabletMargin should render correctly', () => {
    const { container } = render(
      <Container isLayout overrides={{ Container: { tabletMargin: 5 } }}>
        hello world
      </Container>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Container.base should render correctly', () => {
    const { container } = render(<Container>hello world</Container>, {
      theme: { Container: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.fluid should render correctly', () => {
    const { container } = render(<Container isFluid>hello world</Container>, {
      theme: { Container: { styles: { fluid: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.layout should render correctly', () => {
    const { container } = render(<Container isLayout>hello world</Container>, {
      theme: { Container: { styles: { layout: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.fluidMargin should render correctly', () => {
    const { container } = render(<Container isFluid>hello world</Container>, {
      theme: { Container: { fluidMargin: 3 } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Container.tabletMargin should render correctly', () => {
    const { container } = render(<Container isLayout>hello world</Container>, {
      theme: { Container: { tabletMargin: 5 } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Container>hello world</Container>, {
      theme: { Container: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for breakpoint', () => {
    const { container } = render(<Container>hello world</Container>, {
      theme: { Container: { defaultProps: { breakpoint: 'desktop' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for isFluid', () => {
    const { container } = render(<Container>hello world</Container>, {
      theme: { Container: { defaultProps: { isFluid: true } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for isLayout', () => {
    const { container } = render(<Container>hello world</Container>, {
      theme: { Container: { defaultProps: { isLayout: true } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
