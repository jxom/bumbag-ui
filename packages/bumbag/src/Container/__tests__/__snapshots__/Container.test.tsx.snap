// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  Hello world
</p>
`;

exports[`composition hook should return with Container props 1`] = `
Object {
  "className": "css-1fds1n css-1pwrqjj css-1yuzjtp fp-Box fp-Container",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for breakpoint 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  max-width: 1024px;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`defaultProps should render correctly for isFluid 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`defaultProps should render correctly for isLayout 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`overrides Container.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

.emotion-0 {
  background-color: #ff0000;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`overrides Container.fluid should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem;
}

.emotion-0 {
  background-color: #ff0000;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`overrides Container.fluidMargin should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  padding-left: 3rem;
  padding-right: 3rem;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`overrides Container.layout should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-0 {
  background-color: #ff0000;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`overrides Container.tabletMargin should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 5rem;
    padding-right: 5rem;
  }
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`props alignment renders correctly for a center aligned container 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props alignment renders correctly for a left aligned container 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props alignment renders correctly for a right aligned container 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props fixed width renders correctly when breakpoint is desktop 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  max-width: 1024px;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props fixed width renders correctly when breakpoint is fullHD 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  max-width: 1440px;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props fixed width renders correctly when breakpoint is mobile 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  max-width: 480px;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props fixed width renders correctly when breakpoint is tablet 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  max-width: 768px;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props fixed width renders correctly when breakpoint is widescreen 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  max-width: 1200px;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props fluid renders correctly for a fluid container 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props layout renders correctly for a layout container 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  test
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

.emotion-2 {
  color: #574feb !important;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  Hello world
</div>
`;

exports[`theming Container.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

.emotion-0 {
  background-color: #ff0000;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`theming Container.fluid should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem;
}

.emotion-0 {
  background-color: #ff0000;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`theming Container.fluidMargin should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  padding-left: 3rem;
  padding-right: 3rem;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`theming Container.layout should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

.emotion-0 {
  background-color: #ff0000;
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;

exports[`theming Container.tabletMargin should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:768px) {
  .emotion-0 {
    padding-left: 5rem;
    padding-right: 5rem;
  }
}

@media (max-width:1568px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-0 {
    max-width: 768px;
  }
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Container"
>
  hello world
</div>
`;
